### ISSUE TITLE
Build status on initial GH commit in Pending state

### ISSUE THREAD

**Issue by bartoszmajsak** [openshiftio/openshift.io#1259]
After creating new quickstart in OSIO I end up with initial commit on GH being in "pending build state" forever. 

![image](https://user-images.githubusercontent.com/719616/32269168-e95a971c-bef1-11e7-829d-c3c3deec015d.png)

Clicking on it gets me to `http://unconfigured-jenkins-location/job/bartoszmajsak/job/bartosz-sdp-test-01/job/master/2/display/redirect`.

**Comment by bartoszmajsak** [comment]
Out of curiosity - why the commit says `openshift-io committed with jboss `? Where is this jboss thingy coming from?

```
$ git show
commit 99588e832d7f4d76decb84a9d8507f2a14bb7d65 (HEAD -> master, origin/master, origin/HEAD)
Author: bartoszmajsak <openshiftio@redhat.com>
Date:   Wed Nov 1 09:21:31 2017 +0000

    Initial import
```

**Comment by kbsingh** [comment]
unsure how the service-delivery team can help here ? removing tag, but let us know if there is anything we can help with . this looks to be a cicd team / forge area problem.

**Comment by bartoszmajsak** [comment]
This also leads to losing status of the PR and be able to merge even when builds fails, as there is no update fed back to GH.

**Comment by bartoszmajsak** [comment]
Obviously, the same problem occurs on PR checks, as status API from GH is related to any commits, including merge commits which are shown in PR status.

Bumping it to SEV2 as this should be in my opinion fixed rather urgently. cc @chmouel @lordofthejars

**Comment by bartoszmajsak** [comment]
Duplicate: https://github.com/openshiftio/openshift.io/issues/2504

**Comment by sthaha** [comment]
### Steps to reproduce
1. create quickstart project
2. create a pull-request to the project on GitHub
3. wait for the CI to trigger
4. Look at the `Details` link see: ![ci-run-link](https://user-images.githubusercontent.com/3005132/37133707-b43ebc68-22e0-11e8-8fb0-0ddd8a722397.png)

### Expected
Must point to actual jenkins job console

### Actual 
Points to `http://unconfigured-jenkins-location/job/{user}/job/{repo}/job/{branch}/1/display/redirect`

**Comment by chmouel** [comment]
I add this to our backlog on https://github.com/orgs/fabric8io/projects/2

**Comment by piyush1594** [comment]
I am able to reproduce both the issue @bartoszmajsak and @sthaha has mentioned.

**Comment by sthaha** [comment]
This would only be solved as a part of CICD.next as link to jenkins.openshift.io can only route to the jenkins of the logged in user.

Perhaps we should close this as won't fix

/cc @krishnapaparaju @pradeepto

**Comment by chmouel** [comment]
Issue reported here by @gabemontero here https://issues.jenkins-ci.org/browse/JENKINS-48079

we need to reopen this @sthaha this is a jenkins image issue, the problem is actually that we don't have a proper jenkins.model.JenkinsLocationConfiguration.xml in $JENKINS_HOME

I have just done this with the right URL, for example : 

```xml
<?xml version='1.0' encoding='UTF-8'?>
<jenkins.model.JenkinsLocationConfiguration>
  <jenkinsUrl>https://jenkins-cboudjna-osiotest1-jenkins.b542.starter-us-east-2a.openshiftapps.com/</jenkinsUrl>
</jenkins.model.JenkinsLocationConfiguration>
```

and I have it reporting properly the URL : 

![image](https://user-images.githubusercontent.com/98980/42934680-ff9f2038-8b47-11e8-9e4c-287c17f2d6fa.png)

Since there is a bug what about we set this up at boot time in jenkins script boot ? We have all the info we want there to get it

**Comment by sthaha** [comment]
Filed in Planner: https://openshift.io/openshiftio/Openshift_io/plan/detail/312

**Comment by bartoszmajsak** [comment]
@sthaha wassup with this one? :)

**Comment by sthaha** [comment]
@bartoszmajsak that has been filed as a feature-request in Planner see the [planner link](https://openshift.io/openshiftio/Openshift_io/plan/detail/312)

Supporting this requires changes to the way proxy handles requests. Currently jenkins.openshift.io forwards requests to **their** jenkins instance. So if **I** submit a PR to **your project** just publishing the link to jenkins.openshift.io/jobs/repo/pr-<num>/ won't help as it will try to take me to **my** jenkins (if available) instead of yours.
